pexec/managed_process.go

```diff
@@ -32,6 +32,10 @@ type ManagedProcess interface {
 	// there's any system level issue stopping the process.
 	Stop() error
 
+	// Kill will attempt to kill the process group and not wait for completion. Only use this if
```

Comment:
What's the story behind killing the process group? Is the `viam-server` part of the same process group? If we're intentionally trying to kill the viam-server + all modules running in one swift process group `kill -9`, what's the utility in adding this functionality to the goutils/managed process code?